### PR TITLE
#181339468 Setting up postgres database and sequelize

### PR DESCRIPTION
**Pull request by friendscareer**
### What does this PR do?

Configure BareFoot Nomad to use PostgreSQL as DB Engine and Sequelize ORM

### Description of Task to be completed?

Create postgres database on Heroku.

Use sequlize as our ORM on the API. Set up the migrations and models structure, have a dummy user table and model setup to test the whole configuration.

### How should this be manually tested?

- After cloning this repository, install all packages on your local machine with **`yarn install`** an thern run the project using **yarn run dev** script.
- Create a **`.env`** file in root directory of your cloned repo and copy variables provided in **`.env.sample`** file; assign **`DEV_DATABASE_URL`** with your local postgres URL to test the connection.

### Background Context

You will need to have database created in your local postgres database before migrating and seeding data, otherwise you will have an error.

NB: Remote database is not working for now. 

### Relevant Pivotal Tracker Stories

Click -> [181339468](https://www.pivotaltracker.com/story/show/181339468)

### Screenshots (if appropriate)
db

### Questions:

For now the project runs well using local postgres database but has issues while connecting to the remote Heroku database. I am still trying to fix it, will update once it is fixed.

### CODE REVIEW

File: src/models/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const fs = require("fs");
+const path = require("path");
```
Comment (Code-250) [inline src/models/index.js]: can you try to use ES6 instead of ES5

File: src/models/user.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+"use strict";
+const { Model } = require("sequelize");
```
Comment (Code-250) [inline src/models/user.js]: try to use ES6 instead of ES5

File: src/database/config/config.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+require("dotenv").config();
```
Comment (Code-250) [inline src/database/config/config.js:1]: ES6 instead of ES5

File: src/app.js [hunk 4]
```diff
@@ -1,12 +1,24 @@
-import express from "express"
-import routes from "routes/index"
-import "dotenv/config"
+import express from "express";
+import routes from "routes/index";
+import "dotenv/config";
+const Sequelize = require("sequelize");
```
Comment (Code-250) [inline src/app.js]: use ES6 instead of ES5

File: .env.sample [hunk 5]
```diff
@@ -0,0 +1 @@
+DEV_DATABASE_URL=<postgress url>
```
Comment (Code-250) [inline .env.sample]: add new line at the end of the file

File: .sequelizerc [hunk 6]
```diff
@@ -0,0 +1,8 @@
+const path = require("path");
```
Comment (Code-250) [inline .sequelizerc:1]: ES6 instead of ES5

**Review by Code-250 — CHANGES_REQUESTED** [review]

File: src/models/user.js (hunk 2 above)
Comment (Eli250) [reply src/models/user.js]: This is an auto-generated code from sequelize. If you know how to convert this line in ES6 please assist.

File: src/app.js (hunk 4 above)
Comment (Eli250) [reply src/app.js]: Resolved

File: src/models/user.js (hunk 2 above)
Comment (Eli250) [reply src/models/user.js]: Resolved

**Pull request closed without merging**